src/main/scala/sigmastate/utxo/transformers.scala

```diff
@@ -211,13 +211,13 @@ case class Fold[IV <: SType](input: Value[SCollection[IV]],
 
 object Fold {
   def sum[T <: SNumericType](input: Value[SCollection[T]])(implicit tT: T) =
-    Fold(input, 21, Constant(tT.upcast(0.toByte), tT), 22, Plus(TaggedVariable(22, tT), TaggedVariable(21, tT)))
+    mkFold(input, 21, Constant(tT.upcast(0.toByte), tT), 22, Plus(mkTaggedVariable(22, tT), mkTaggedVariable(21, tT))).asInstanceOf[Fold[T]]
```

Comment:
@greenhat I don't think we need builders in concrete factory methods.
They are usefull in more generic context, where the builder come as a parameter and can be configured.

Reply:
@ergomorphic Agree. Fixed.